docs/guide.rst

```diff
@@ -1151,12 +1151,34 @@ function:
    myWidget :: ... -> Widget n
    myWidget ... =
        Widget Fixed Fixed $ do
-           ...
+           ... (rendering function) ...
+
+Input arguments to this function will typically be whatever data is
+being maintained as internal state for your Widget, and the output is
+the renderable Widget.  This constructor is called each time brick
+needs to draw output.
 
 We specify the horizontal and vertical growth policies of the widget
 as ``Fixed`` in this example, although they should be specified
-appropriately (see `How Widgets and Rendering Work`_). Lastly we specify
-the *rendering function*, a function of type
+appropriately (see `How Widgets and Rendering Work`_).
+
+If your Widget is made entirely by composing existing Widgets, the
+constructor can be written easily.  For example:
+
+.. code:: haskell
+
+   myWidget :: String -> String -> Widget n
+   myWidget title body =
+       Widget Fixed Fixed $
+           withDefAttr myTitleAttr (str title)
+           <=>
+           str body
+           <=>
+           str "The end."
```

Comment:
In examples like this that don't use the rendering monad at all, the constructor should be omitted. The way to write this particular example is
```
myWidget :: String -> String -> Widget n
myWidget =
    vBox [ withDefAttr myTitleAttr (str title)
         , str body
         , str "The end."
         ]
```
In other words, no explicit constructor is needed because `<=>` constructs a `Widget` for you.